config/class-site-details-index.php

```diff
@@ -228,6 +248,74 @@ public function get_jetpack_info() {
 		return $jetpack_info;
 	}
 
+	/**
+	 * Gather all the information about Parsely
+	 * @return array Parsely plugin info.
+	 */
+	public function get_parsely_info() {
+		$parsely      = get_option( 'parsely' );
+		$parsely_info = [];
+
+		$parsely_info['active']           = false;
+		$parsely_info['integration_type'] = $this->get_parsely_integration_type();
+
+		if ( self::PARSELY_INTEGRATION_TYPE_MUPLUGINS === $parsely_info['integration_type'] ||
+			self::PARSELY_INTEGRATION_TYPE_MUPLUGINS_SILENT === $parsely_info['integration_type'] ||
+			self::PARSELY_INTEGRATION_TYPE_SELF_MANAGED === $parsely_info['integration_type']
+		) {
+			$parsely_info['active'] = true;
+		}
+
+		$parsely_info['parsely_service_type'] = self::PARSELY_SERVICE_TYPE_NONE;
+
+		if ( self::PARSELY_INTEGRATION_TYPE_MUPLUGINS === $parsely_info['integration_type'] ||
+			self::PARSELY_INTEGRATION_TYPE_SELF_MANAGED === $parsely_info['integration_type']
+		) {
+			$parsely_info['parsely_service_type'] = self::PARSELY_SERVICE_TYPE_PAID;
+		}
+
+		$parsely_info['version'] = $this->get_parsely_version( $parsely );
+
+		return $parsely_info;
+	}
+
+	/**
+	 * Gets the Parsely version
+	 * @param array $parsely The Parsely options.
+	 * @return string The version.
+	 */
+	public function get_parsely_version( $parsely ) {
+		if ( ! $parsely || ! isset( $parsely['plugin_version'] ) ) {
+			return self::PARSELY_VERSION_NONE;
+		}
+
+		return $parsely['plugin_version'];
+	}
+
+	/**
+	 * The way that the plugin is used in WordPress
+	 * @return string integration type.
+	 */
+	public function get_parsely_integration_type() {
+		// detect a hidden installation
+		if ( false !== get_option( '_wpvip_parsely_mu', false ) && ! has_filter( 'wpvip_parsely_load_mu' ) ) {
```

Comment:
Rather than using `has_filter` to detect things, I wonder if it would be better to set have the code in `wp-parsely.php` track / share necessary details instead which would we infer the data more directly from.

e.g. could have a static class called `Parsely_Loader_Info` with methods `(get|set)_integration_type`. `wp-parsely.php` would call `Parsely_Loader_Info::set_integration_type( ... )` and here we'd fetch it via `Parsely_Loader_Info::get_integration_type()`.

Reply:
> Rather than using `has_filter` to detect things, I wonder if it would be better to set have the code in `wp-parsely.php` track / share necessary details instead which would we infer the data more directly from.
> 
> e.g. could have a static class called `Parsely_Loader_Info` with methods `(get|set)_integration_type`. `wp-parsely.php` would call `Parsely_Loader_Info::set_integration_type( ... )` and here we'd fetch it via `Parsely_Loader_Info::get_integration_type()`.

Im not opposed to this change but I'm not clear on what problem it is attempting to solve by this. I'm anxious to hear more of your wisdom on this.

Reply:
I have moved most of the logic into the new `Parsely_Loader_info` class in `wp-parsely.php`. It has all of the functionality that you've asked for and will keep these values in static properties so that they only have to be resolved once per property.